internal/gitinterface/changes.go

```diff
@@ -36,17 +37,57 @@ func GetCommitFilePaths(commit *object.Commit) ([]string, error) {
 
 // GetFilePathsChangedByCommit returns the paths changed by the commit relative
 // to its parent commit. If the commit is a merge commit, i.e., it has more than
-// one parent, no changes are returned.
-//
-// Currently, this function does not verify that the tree for a merge commit
-// matches one of its parents. In a future version, this behavior may change and
-// return an error if a multi-parent commit seems invalid.
+// one parent, check if the commit is the same as at least one of its parents.
+
 func GetFilePathsChangedByCommit(repo *git.Repository, commit *object.Commit) ([]string, error) {
 	if len(commit.ParentHashes) > 1 {
-		// merge commits are expected not to introduce changes themselves
-		// TODO: should we check that the merge commit's tree matches one of its
-		// parents (usually the last)?
-		return nil, nil
+		// Merge commit: compare with each parent and aggregate changes
+
+		// Create a map to store all changes
+
+		contains := make(map[string]bool)
+
+		// keeping a heap of diffs so that we can pop them in sorted order
+
+		diffs := &diffHeap{}
+		heap.Init(diffs)
+
+		// We are iterating backwards since if their is a matching parent commit,
+		// it is likely to be the last one.
+		for parentHashIndex := len(commit.ParentHashes) - 1; parentHashIndex >= 0; parentHashIndex-- {
+			parentCommit, err := GetCommit(repo, commit.ParentHashes[parentHashIndex])
+			if err != nil {
+				return nil, err
+			}
+
+			// Get the diff file paths between the commit and its parent
```

Comment:
Since you have the current commit and parent commit, we can return early here if the tree IDs match. We don't have to fetch both trees etc (which is what happens down that call stack) and then return on `len(diff) == 0`.

Reply:
I have added a check before checking the diffs to check if the commit and parent tree hash are equal.